docs/lake/design.md

```diff
@@ -0,0 +1,658 @@
+our goal here is that the Zed lake tooling be as easy and familiar as git is
+to a technical user.
+
+While this design document is independent of any particular implementation,
+we will illustrate examples of `zed lake` commands that are under development.
+Note that while this CLI-first approach provides an ergonomic way to experiment with
+and learn the Zed lake building blocks, all of this functionality is also
+exposed through an API to a cloud-based service.  Most interactions between
+a user and a Zed lake would be via an application like
+[Brim](https://github.com/brimdata/brim) or a
+programming environment like Python/Pandas rather than via direct interaction
+with `zed lake`.
+
+### New
+
+A new pool is created with
+```
+zed lake new -p <name> -k <key>[,<key>...]
+```
+where `<name>` is the name of the pool within the implied lake instance and
+`<key>` is the Zed language representation of the pool key, e.g.,
+```
+zed lake new -p logs -k ts
+```
+Note that there may be multiple pool keys, where subsequent keys act as the secondary,
+tertiary, and so forth sort key.
+
+### Load
+
+Data is then loaded into a lake with the `load` command, .e.g.,
+```
+zed lake load -p logs sample.ndjson
+```
+where `sample.ndjson` contains logs in NDJSON format.  Any supported format
+(i.e., CSV, NDJSON, Parquet, ZNG, and ZST) as well multiple files can be used
+here, e.g.,
+```
+zed lake load -p logs sample1.csv sample2.zng sample3.zng
+```
+Parquet and ZST formats are not auto-detected so you must currently specify
+`-i` with these formats, e.g.,
+```
+zed lake load -p logs -i parquet sample4.parquet
+zed lake load -p logs -i zst sample5.zst
+```
+
+Note that there is no need to define a schema or insert data into
+a "table" as all Zed data is _self describing_ and can be queried in a
+schema-agnostic fashion.  Data of any _shape_ can be stored in any pool
+and arbitrary data _shapes_ can coexist side by side.
+
+### Scan
+
+Data is read from a pool with the `scan` command.  By default, `scan`
+generates sorted output of all of the pool data in the configured key order.
+The order can be overridden with `-order asc` or `-order desc`.
+
+A range can be specified with `-from` or `-to`.
+
+The default output format is ZNG though it can be overridden with the various
+supported output formats.
+
+This example reads every record from the `logs` pool starting
+from time `2020-1-1T12:00`, in ascending order,
+sends the  results as ZNG to stdout, then pipes the output to `zq` to count the records:
+```
+zed lake scan -p logs -from 2020-1-1T12:00 -order asc | zq -z "count()" -
+```
+
+### Query
+
+Of course, the example is an inefficient way to count records.  Instead of
+reading the records out of the lake and processing them, a better approach
+is to push the query into the lake.
+
+The `query` command lets you do this, e.g.,
+```
+zed lake query -p logs -from 2020-1-1T12:00 -order asc "count()" -
```

Comment:
```suggestion
zed lake query -p logs -from 2020-1-1T12:00 -order asc "count()"
```

Oops. I screwed up with my other review comment. This was the place where I intended to flag what I thought was a misplaced trailing `-`.